main.go

```diff
@@ -31,9 +47,59 @@ func init() {
 }
 
 func main() {
+	log.Namespace = namespace
 	ctx := context.Background()
-	if err := service.Run(ctx, taxonomyRedirects); err != nil {
-		log.Event(nil, "unable to run application", log.Error(err), log.FATAL)
+
+	// Get Config
+	cfg, err := config.Get()
+	if err != nil {
+		log.Event(ctx, "unable to retrieve service configuration", log.FATAL, log.Error(err))
+		os.Exit(1)
+	}
+	log.Event(ctx, "got service configuration", log.Data{"config": cfg}, log.INFO)
+
+	// Make os Signal channels
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+
+	// Create healthcheck with versionInfo
+	versionInfo, err := healthcheck.NewVersionInfo(BuildTime, GitCommit, Version)
+	if err != nil {
+		log.Event(ctx, "failed to create versionInfo for healthcheck", log.FATAL, log.Error(err))
+		os.Exit(1)
+	}
+	hc := healthcheck.New(versionInfo, cfg.HealthCheckCriticalTimeout, cfg.HealthCheckInterval)
+
+	hc.Start(ctx)
+
+	// Run the HTTP service
+	httpServer := service.Run(ctx, cfg, taxonomyRedirects, &hc)
+
+	// Wait for OS signal, and do graceful shutdown
+	signal := <-signals
+	log.Event(ctx, "shutting down after os signal received", log.INFO, log.Data{"signal": signal, "shutdown_timeout": cfg.ShutdownTimeout})
+	shutdownCtx, cancel := context.WithTimeout(context.Background(), cfg.ShutdownTimeout)
+
+	// Shutdown goroutine
+	go func() {
+		log.Event(shutdownCtx, "stopping healthcheck", log.INFO)
+		hc.Stop()
+
+		log.Event(shutdownCtx, "stopping http server", log.INFO)
+		if err := httpServer.Shutdown(shutdownCtx); err != nil {
+			log.Event(shutdownCtx, "failed to gracefully shutdown http server", log.FATAL, log.Error(err))
+			os.Exit(1)
```

Comment:
I wouldn't exit here but capture that the graceful shutdown has failed, again see [dataset API pr](https://github.com/ONSdigital/dp-dataset-api/pull/215)